feat: add direct ITRS↔Observed transforms | swev-id: astropy__astropy-13398

casey-brooks opened the pull request:
## Summary
- add dedicated itrs_observed_transforms module with direct transforms
- add coverage for round-trip, unit-spherical, broadcasting, and warning cases
- document new functionality and limitations in common errors guide

## Testing
- LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ./.venv/bin/python run_pytest.py tmp_pytest/test_itrs_observed_transforms.py -q
- LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ./.venv/bin/python run_pytest.py tmp_pytest/test_intermediate_transformations_wrapper.py -q
- ./.venv/bin/ruff check astropy/coordinates/builtin_frames/itrs_observed_transforms.py astropy/coordinates/tests/test_itrs_observed_transforms.py

Fixes #70

casey-brooks committed:
feat(coordinates): add direct itrs observed transforms

casey-brooks commented:
### Test & Lint Summary
- LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ./.venv/bin/python run_pytest.py tmp_pytest/test_itrs_observed_transforms.py -q (10 passed)
- LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ./.venv/bin/python run_pytest.py tmp_pytest/test_intermediate_transformations_wrapper.py -q (1 passed)
- ./.venv/bin/ruff check astropy/coordinates/builtin_frames/itrs_observed_transforms.py astropy/coordinates/tests/test_itrs_observed_transforms.py (no issues)

rowan-stein commented:
Requesting review: Direct ITRS<->Observed transforms within ITRS (ITRS<->AltAz, ITRS<->HADec).

Please focus on:
- Geometric correctness (ENU basis, handedness, azimuth convention)
- Obstime mismatch warning behavior (no ITRS time propagation)
- Unit-spherical handling and distance requirements
- Refraction ignored warning and docs alignment
- Test coverage (round-trips, zenith, broadcasting, warnings, direct vs ICRS comparison)
- Documentation updates and warning/error message clarity

References:
- Issue #70: Direct ITRS<->Observed transforms within ITRS
- Task swev-id: astropy__astropy-13398

casey-brooks commented:
Addressed review feedback:
- obstime mismatch warning now compares Time differences with 1e-9 s tolerance and no blanket exception handling
- pressure warning requires convertible Quantity and lets conversion errors propagate
- Observed->ITRS distance error message clarifies finite distance requirement

Re-ran targeted checks:
- ASTROPY_USE_IERS_A=never ASTROPY_DOWNLOAD_IERS_DATA=0 LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ../../../bin/pytest astropy/coordinates/tests/test_itrs_observed_transforms.py -q (10 passed)
- ASTROPY_USE_IERS_A=never ASTROPY_DOWNLOAD_IERS_DATA=0 LD_LIBRARY_PATH=/nix/store/wffgswxkp55xi14jy63rjsnfvl2qvmxy-gcc-14.3.0-lib/lib ../../../bin/pytest astropy/coordinates/tests/test_intermediate_transformations.py::test_straight_overhead -q (1 passed)
- .venv/bin/ruff check astropy/coordinates/builtin_frames/itrs_observed_transforms.py (clean)

Note: repository rules blocked direct pushes to feature/itrs-observed-direct-astropy-13398. I pushed the update to feature/itrs-observed-direct-astropy-13398-fix and opened PR #87 targeting this branch so the changes can be merged via an approved PR.
